### ISSUE TITLE
Deletes all files if we try to git commit without staging files

### ISSUE THREAD

**Issue by deepakmahakale** [sds/overcommit#751]
overcommit deletes all files if we try to commit without adding the files

Steps to reproduce:

```
1. Change a file
2. Do not `git add`
3. Try to commit 
    git commit -m "commit message"

4. Notice all the files have been deleted and staged 
```

The issue can be reproduced on this repository:

https://github.com/deepakmahakale/overcommit_issue

**Comment by sds** [comment]
Thanks for reporting this issue.

This is strange. It appears to be a change in the behavior of `git stash`, or has somehow always been present but never noticed. You can reproduce with pure Git (i.e. with no direct use of Overcommit) using the following steps:

1. Modify a file in your repository. Do **not** add via `git add ...`.
2. Run `git -c commit.gpgsign=false stash save --keep-index --quiet`
3. Confirm your working directory is free of any unstaged changes.
4. Run `git stash pop --index --quiet`.

This reproduces the behavior.

This _seems_ like a bug with `git stash`, but perhaps we are missing something. Would appreciate any help in tracking down the underlying problem, as this seems like a very straightforward use of `git stash` that isn't working as expected.

**Comment by deepakmahakale** [comment]
I had something in mind but I haven't tried it yet.

Let's say if we just add an additional check/hook that checks for at least 1 staged file before commit. 
Will see if something like this works

**Comment by deepakmahakale** [comment]
@sds I have opened a draft pull request. 
Adding a check for staged files seems to be working.

**Comment by sds** [comment]
Fixed in 648383ff3cd9d51ab83de87f2bf64fa9a0d6d559.

**Comment by deepakmahakale** [comment]
Thanks @sds .
